File: 2917-count-pairs-whose-sum-is-less-than-target/solution.java
class Solution {
    public int countPairs(List<Integer> nums, int target) {
        int i=0;
        int j=nums.size()-1;
        int count = 0;

        Collections.sort(nums);
        while(i < j) {
            if(nums.get(i) + nums.get(j) < target) {
                count += j-i; // include whole window
                i++;
            } else {
                j--;
            }
        }
        return count;
    }
}

File: 2917-count-pairs-whose-sum-is-less-than-target/solution.swift
class Solution {
    func countPairs(_ nums: [Int], _ target: Int) -> Int {
        var arr = nums
        arr.sort()
        var n = arr.count
        var i=0
        var j=n-1
        var count = 0
        while(i < j) {
            if(arr[i] + arr[j] < target) {
                count += j-i
                i += 1
            } else {
                j -= 1
            }
        }
        return count
    }
}
